java/src/main/java/com/google/appengine/tools/pipeline/JobRunId.java

```diff
@@ -0,0 +1,94 @@
+package com.google.appengine.tools.pipeline;
+
+import com.google.appengine.tools.pipeline.impl.model.JobRecord;
+import com.google.cloud.datastore.Key;
+import com.google.common.base.Preconditions;
+import lombok.*;
+
+import javax.annotation.Nullable;
+import java.io.Serial;
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * identifies a particular run of a job in the pipelines framework
+ *  also, a root job, which identifies a run a *pipeline*
+ *
+ */
+@EqualsAndHashCode(callSuper = false)
+@AllArgsConstructor(access = AccessLevel.PROTECTED)
+@Getter
+public class JobRunId implements Serializable {
+
+  public static final String DELIMITER = ":";
+
+  @Serial
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * project in which job is executing
+   */
+  @NonNull
+  private final  String project;
```

Comment:
indent? (in the file, not only here)
```suggestion
  private final String project;
```